Fix #15149: [CD] Add reviewer name to reviewed suggestion modal card

Pratik960 opened the pull request:
1. This PR fixes or fixes part of #15149 .
2. This PR does the following: Added Reviewer Name to to reviewed suggestion modal card so that contributor can see who reviewed their suggestion


## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct
![added reviewername](https://user-images.githubusercontent.com/77715255/223108220-9b66cf73-fe38-4e1d-96aa-ecc364b470ad.png)


#### Proof of changes on mobile phone

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).

Pratik960 committed:
Fix #15149 [CD] Add reviewer name to reviewed suggestion modal card

Added Reviewer Name to to reviewed suggestion modal card so that contributor can see who reviewed their suggestion

Pratik960 committed:
Merge branch 'develop' into pratikupdate

Pratik960 committed:
Fix #15149: [CD] Add reviewer name to reviewed suggestion modal card

There were some errors in previous pull request so solved that errors

Pratik960 committed:
Merge branch 'pratikupdate' of https://github.com/Pratik960/oppia into pratikupdate

kevintab95 reviewed (changes requested):
Hi @Pratik960, if local server is not working, how were these changes tested?

Pratik960 commented:
@kevintab95  Could you please guide me on what to do when I need to test changes, but the server is not running due to a memory issue? is there any other way for testing changes locally?

Pratik960 committed:
FIx #17512 [CD] Add reviewer name to reviewed suggestion modal card

Edited test cases

Pratik960 commented:
@kevintab95 Can you suggest me what should i do?

sagangwee commented:
@Pratik960 What is the memory issue that you are facing? Can you post a screenshot?

Pratik960 committed:
Reviewer Name error solving

The contrubuter is seeing own username as  reviewername  it is fixed in this commit

Pratik960 committed:
Merge branch 'oppia:develop' into pratikupdate

Pratik960 commented:
@sagangwee Sir the issue is solved now, i have asked on oppia discussion and got answer on how to solve this issue.

Pratik960 commented:
@sagangwee @kevintab95 Sir i am not getting reviewer name in the contributor dashboard of user i think the error is in this code  in file 
core\templates\pages\contributor-dashboard-page\modal-templates\question-suggestion-review-modal.component.ts
![code](https://user-images.githubusercontent.com/77715255/223012448-5aef45a8-0007-4c6c-9515-258a4031551c.png)
i have defined reviewername property in
core\templates\domain\feedback_message\ThreadMessage.model.ts
![threadmessage](https://user-images.githubusercontent.com/77715255/223013116-7037e0df-64c3-48b6-9ea0-77e23ffbc247.png)

The result is 
In admin page reviewername is entered but not able to see in user review page i have tried all the possible solution i came up with but still not getting reviewer name could you help me with it sir?
![image](https://user-images.githubusercontent.com/77715255/223018526-6c3d5353-42de-414f-ba90-c2739a21249b.png)

Pratik960 commented:
@kevintab95 @sagangwee  PTAL!!!!

sagangwee commented:
@Pratik960 A few things:
1. The "Proof that changes are correct" shows a blank box for the reviewer name. Can you also show a screenshot of what it looks like when the name is populated?
2. I would make the label just "Reviewer:".
3. Let's not have a text box for the reviewer name since that implies user input and just display the name as text. 
4. The reviewer name should only be displayed when users are looking at their own accepted/rejected submission. Please also a screenshot when the submission is accepted or rejected.

Thanks!

kevintab95 reviewed (commented):
Hi @Pratik960, please feel free to assign me once the changes requested have been completed. Thanks!
